### PR TITLE
Allow passing cap to VideoCapture constructor

### PR DESCRIPTION
**Pull request by la-we**
Currently working on a project where i need functionality of passing explicit `CAPs` to `VideoCapture`, figured i contribute a little.

### CODE REVIEW

**Comment by UrielCh** [comment]
Please try to match parameters names from openCV:

[VideoCapture](https://docs.opencv.org/4.10.0/d8/dfe/classcv_1_1VideoCapture.html#ac4107fb146a762454a8a87715d9b7c96)(const String &filename, int apiPreference=[CAP_ANY])

Opens a video file or a capturing device or an IP video stream for video capturing with API Preference.


the parametere is named apiPreference, not cap

File: cc/io/VideoCapture.cc [hunk 1]
```diff
@@ -27,11 +27,19 @@ NAN_METHOD(VideoCapture::New) {
   FF::TryCatch tryCatch("VideoCapture::New");
   FF_ASSERT_CONSTRUCT_CALL();
   VideoCapture* self = new VideoCapture();
+
+  uint32_t apiPreference = cv::CAP_ANY;
+  if (info.Length() == 2) {
```
Comment (UrielCh) [inline cc/io/VideoCapture.cc:32]: the latest vertion allow a 3th parameter, so the test should be >=2

and is you want a more cleaner code throw an not implemented yet if more than 2 parameters are provided.

the 3th parameter is only available in openCV ~~ 4.5+

File: test/pnpm-lock.yaml [hunk 2]
```diff
@@ -1,967 +1,637 @@
-lockfileVersion: '6.0'
+lockfileVersion: '9.0'
```
Comment (UrielCh) [inline test/pnpm-lock.yaml:1]: - please avoid commiting generated file.
- an upgraded pnpm lock file will make fail all the CI.

please roll this file back to make the CI work.

or upgrate all workflow to make the CI back.

Comment (UrielCh) [reply test/pnpm-lock.yaml:1]: if the CI pass, you are lucky.

**Comment by UrielCh** [comment]
I live you check why some CI tack fail.

File: test/pnpm-lock.yaml (hunk 2 above)
Comment (la-we) [reply test/pnpm-lock.yaml]: Reverted to previous version

File: cc/io/VideoCapture.cc (hunk 1 above)
Comment (la-we) [reply cc/io/VideoCapture.cc]: New changes should cover all constructor variant a user might try to use and throw if it is not yet implemented

**Comment by la-we** [comment]
Tests should pass now on all OSes

**Comment by UrielCh** [comment]
the test still failed on macos.

**Comment by UrielCh** [comment]
LGTM

**Pull request merged**